internal_ws/yktrace/src/swt.rs

```diff
@@ -29,7 +29,8 @@ impl ThreadTracerImpl for SWTThreadTracer {
     }
 }
 
-pub(crate) fn start_tracing() -> ThreadTracer {
```

Comment:
I'm not sure if this is right. Why would starting tracing be a testing feature?

It's probably because your tool build with hardware tracing rustflags (and not software tracing)?

Reply:
Similarly for the other bits in this file.

Reply:
> It's probably because your tool build with hardware tracing rustflags (and not software tracing)?

Yes, I used `hw`. I didn't realise that there'd be a difference here with `sw`? Would a bors try flush this out?

Reply:
> I didn't realise that there'd be a difference here with sw?

My guess is that compile-time guards like [this](https://github.com/softdevteam/yk/blob/6023c7c091f2483f5a22344a73be4575ad2be254/internal_ws/yktrace/src/lib.rs#L74) would give inaccurate results if the oracle uses only `-C tracer=hw`.

Ideally your oracle would include both kinds of tracing, but read on...

> Would a bors try flush this out?

No, because we don't yet test software tracing with cranelift. I'm not sure it's ready yet, but @bjorn3 is the person to ask.

Reply:
Having looked at this again, I agree that applying this to `swt.rs` is incorrect. Fixed in https://github.com/softdevteam/yk/pull/309/commits/0703d9473e51a224d08fed953e70b32638b284f0.